staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -32,158 +34,302 @@ import (
 	"k8s.io/klog/v2"
 )
 
-type uniformScenario []uniformClient
+// fairAlloc computes the max-min fair allocation of the given
+// capacity to the given demands (which slice is not side-effected).
+func fairAlloc(demands []float64, capacity float64) []float64 {
+	lim := len(demands)
+	indices := make([]int, lim)
+	for i := 0; i < lim; i++ {
+		indices[i] = i
+	}
+	sort.Slice(indices, func(i, j int) bool { return demands[indices[i]] < demands[indices[j]] })
+	alloc := make([]float64, lim)
+	var next int
+	var prevAlloc float64
+	for ; next < lim; next++ {
+		idx := indices[next]
+		demand := demands[idx]
+		if demand <= 0 {
+			continue
+		}
+		hyp := float64(lim-next) * (demand - prevAlloc)
```

Comment:
I can't figure out what `hyp` stands for...

Reply:
"hypothesis" ... the total usage that would happen if this one gets all it asks for.

Reply:
I gave this variable a more expansive name and an explanatory comment.  I also commented the loop invariant.

Reply:
Thanks, it's much more understandable now.